packages/components/src/Dialog/Dialog.tsx

```diff
@@ -1,52 +1,28 @@
-import {
-  Children,
-  HTMLAttributes,
-  ReactElement,
-  ReactNode,
-  cloneElement,
-  isValidElement,
-  useRef,
-} from 'react';
-
-import { useButton } from '@react-aria/button';
-import { useDialog } from '@react-aria/dialog';
-
-import type { AriaDialogProps } from '@react-types/dialog';
+import { useContext } from 'react';
+import { Dialog, OverlayTriggerStateContext } from 'react-aria-components';
+import type RAC from 'react-aria-components';
 
 import { cn, useClassNames } from '@marigold/system';
 
-import { Header } from '../Header';
-import { Headline } from '../Headline';
-import { DialogContextProps, useDialogContext } from './Context';
 import { DialogController } from './DialogController';
 import { DialogTrigger } from './DialogTrigger';
 
 // Close Button
 // ---------------
 interface CloseButtonProps {
   className?: string;
+  close: () => void;
 }
 
-const CloseButton = ({ className }: CloseButtonProps) => {
-  const ref = useRef<HTMLButtonElement>(null);
-  const { close } = useDialogContext();
-
-  const { buttonProps } = useButton(
-    {
-      onPress: () => close?.(),
-    },
-    ref
-  );
-
+const CloseButton = ({ className, close }: CloseButtonProps) => {
```

Comment:
I think this should use the dialog context and not accept a close method, like here:

https://react-spectrum.adobe.com/react-aria/Dialog.html#state